src/picterra/client.py

```diff
@@ -77,25 +96,7 @@ def upload_raster(self, filename, name):
         resp = self.sess.post(self._api_url('rasters/%s/commit/' % raster_id))
         if not resp.ok:
             raise APIError(resp.text)
-        poll_interval = resp.json()['poll_interval']
-
-        # Wait for raster to be processed
-        def _is_ready():
-            logger.info('checking upload status')
-            resp = self.sess.get(
-                self._api_url('rasters/%s/' % raster_id)
-            )
-            if not resp.ok:
-                logger.warning('failed to get raster status - retrying')
-                return False
-
-            if resp.json()['status'] == 'ready':
-                return True
-            elif resp.json()['status'] == 'failed':
-                raise APIError(resp.text)
-            else:
-                return False
-        _poll_at_interval(_is_ready, poll_interval)
+        self._wait_until_operation_completes(resp.json())
```

Comment:
The commit endpoint for raster now returns an operation ?

Reply:
Yes, well that was in https://gitlab.com/picterra/platform/-/merge_requests/2232, so it's going to be live next week.

Reply:
Thanks !